### PR TITLE
Upgrade ruby and rails versions to latest

### PR DESCRIPTION
**Pull request by axelsson**

### CODE REVIEW

**Comment by axelsson** [comment]
[Update Ruby and rails versions](https://trello.com/c/yMbwpHBB/775-update-ruby-and-rails-versions)

**Review by nanditagoclimate — APPROVED** [review]
🏆

File: Gemfile [hunk 1]
```diff
@@ -1,20 +1,20 @@
 # frozen_string_literal: true
 
 source 'https://rubygems.org'
-ruby '2.6.6'
+ruby '3.1.2'
```
Comment (kallenilver) [inline Gemfile:4]: Tjohooo!

File: config/environments/production.rb [hunk 2]
```diff
@@ -29,26 +29,25 @@
...
 
-  # Mount Action Cable outside main process or domain
+  # Mount Action Cable outside main process or domain.
   # config.action_cable.mount_path = nil
-  # config.action_cable.url = 'wss://example.com/cable'
-  # config.action_cable.allowed_request_origins = [ 'http://example.com', /http:\/\/example.*/ ]
+  # config.action_cable.url = "wss://example.com/cable"
+  # config.action_cable.allowed_request_origins = [ "http://example.com", /http:\/\/example.*/ ]
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
```
Comment (kallenilver) [inline config/environments/production.rb:46]: Why not force ssl in prod?

File: public/i18n/de.json [hunk 3]
```diff
@@ -186,7 +186,7 @@
         },
         "x_years": {
           "one": "ein Jahr",
-          "other": "%{count} Jahr"
+          "other": "%{count} Jahre"
```
Comment (kallenilver) [inline public/i18n/de.json:189]: This seems correct, but also a bit missplaced in this commit?

File: public/i18n/sv.json [hunk 4]
```diff
@@ -708,8 +708,8 @@
       },
       "percentage": {
         "format": {
-          "delimiter": "",
-          "format": "%n%"
+          "delimiter": " ",
+          "format": "%n %"
```
Comment (kallenilver) [inline public/i18n/sv.json:712]: I reluctantly approve this (I've used it the other way all my life but google tells me I'm wrong 😢 )

**Review by kallenilver — APPROVED** [review]
Nice work! Left a few comments, but no show stopper.

File: public/i18n/de.json (hunk 3 above)
Comment (axelsson) [reply public/i18n/de.json:189]: These json files are generated from the yml-files, the last time someone updated the yml-files they we just not updated so these are not changes that I've decided to do in this PR.

File: config/environments/production.rb (hunk 2 above)
Comment (axelsson) [reply config/environments/production.rb:46]: Good that you noticed, should be changed!

**Pull request merged**